src/csharp/Grpc.Examples/MathGrpc.cs

```diff
@@ -287,6 +287,17 @@ protected override MathClient NewInstance(ClientBaseConfiguration configuration)
           .AddMethod(__Method_Sum, serviceImpl.Sum).Build();
     }
 
+    /// <summary>Register service method implementations with a service binder. Useful when customizing the service binding logic.</summary>
```

Comment:
nit: I think the usage of this would be a little more obvious, just from looking at it, if the "default binder" i.e. the `grpc::ServerServiceDefinition::Builder` itself was just the default implementation of a more general "service binder" interface (perhaps the interface could be named something like `ServiceDefinitionBuilder`). Then, the implementation of `grpc::ServerServiceDefinition BindService` would be something like this:

```
public static grpc::ServerServiceDefinition BindService(MathBase serviceImpl)
{
    grpc::ServerBinderBase builder = grpc::ServerServiceDefinition.CreateBuilder();
    BindService(builder, serviceImpl);
    return builder;
}
```

I think this would make it more obvious that "`public static grpc::ServerServiceDefinition BindService` is probably the one you want to use" - because the other `BindService`, which takes a `ServiceBinderBase`, is now just the "thing to use if you want to dip down and override the default builder". What do you think?

Reply:
I thought of that, but it seemed problematic for `grpc::ServerServiceDefinition::Builder` to inherit from `ServiceBinderBase`:
- I'd need to change methods in Builder from non-virtual to virtual which I think is a breaking change.
- I'm not sure if adding a base class to an existing class is fully backwards compatible.
- The AddMethod methods actually return Builder instance, which makes things a bit more complicated.

Another approach would be to keep the existing AddMethod() methods in Builder and add new methods which override the ones declared by ServiceBinderBase, but that's only possible if ServiceBinderBase is an interface (without explicit impl, there is a name clash) and we don't want ServiceBinderBase to be an interfaces as that is risky for future backward compatibility (no default methods implementations for interfaces).
Overall, I'm open to changing, but I'm not sure how to do things cleanly other than with the approach currently used in the PR.